### PR TITLE
Flex 368 runtime

### PR DESCRIPTION
**Pull request by DennyGD**
Add **runtime** option (values: 'node6', 'node8', 'node10') for **flex create**, **flex deploy**, **flex update**.

Note: Based on branch cli-143-env-vars. Probably, it's a good idea to review after PR 134 gets merged.

### CODE REVIEW

File: lib/flex/flexRouter.js [hunk 1]
```diff
@@ -114,6 +130,20 @@ const cmdDefinitions = [
     handlerName: 'logs',
     requirements: [CommandRequirement.AUTH]
   },
+  {
+    command: SubCommand[Namespace.FLEX].UPDATE,
+    desc: 'Update environment variables and runtime',
```
Comment (dokov) [inline lib/flex/flexRouter.js]:
```suggestion
    desc: 'Update environment variables and runtime. Causes restart/rebuild of the service',
```

File: lib/flex/FlexController.js [hunk 2]
```diff
@@ -58,9 +59,49 @@ class FlexController extends BaseController {
...
+    try {
+      if (options[FlexOptionsNames.ENV_VARS_SET]) {
+        this._metadata[FlexOptionsNames.ENV_VARS_SET] = transform(options[FlexOptionsNames.ENV_VARS_SET]);
+      }
+
+      if (options[FlexOptionsNames.ENV_VARS_REPLACE]) {
+        this._metadata[FlexOptionsNames.ENV_VARS_REPLACE] = transform(options[FlexOptionsNames.ENV_VARS_REPLACE]);
+      }
+    } catch (ex) {
+      throw new Error('Environment variables must be specified as comma-separated list where key=value or in valid JSON format.');
```
Comment (dokov) [inline lib/flex/FlexController.js]:
```suggestion
      throw new Error('Environment variables must be specified as comma-separated list (ex. key1=value1,key2=value2) or in valid JSON format.');
```

File: lib/Utils.js [hunk 3]
```diff
@@ -366,6 +366,44 @@ Utils.mapFromSource = function mapFromSource(mapping, source) {
...
+ * Returns an object from a string of key-value pairs where key1[secondaryDelimiter]value[primaryDelimiter] (e.g key1=value1,key2=value2).
+ * Throws an error if pattern is not valid.
+ * @param {String} dsv Delimiter separated key-values.
+ * @param {String} primaryDelimiter Delimiter that separates key-value pairs.
+ * @param {String} secondaryDelimiter Delimiter that separates keys from values.
+ * @returns {Object}
+ */
+Utils.getObjectFromDelimiterSeparatedKeyValuePairs = function getObjectFromDelimiterSeparatedKeyValuePairs(dsv, primaryDelimiter = ',', secondaryDelimiter = '=') {
+  if (typeof dsv !== 'string') {
+    throw new Error('Expected string.');
```
[the remark this answers is not in the document]
Comment (DennyGD) [reply lib/Utils.js]: 'Environment variables must be specified as comma-separated list where key=value or in valid JSON format.' - this is the error that appears as output in the end. 'Cause the CLI has no way of knowing if the user intended to pass JSON and messed it up or they intended to use the shorthand syntax and again messed it up.

[the remark this answers is not in the document]
Comment (dokov) [reply lib/Utils.js]: Ah, so this error can actually happen when the input is actually JSON and then it is not really an error. And it is never shown to customer or logged. OK, I'm good then.

**Comment by DennyGD** [comment]
Comments have been addressed. I'll close this PR and re-open it to master as the `next` branch (the destination branch here) has already been merged into master.

**Pull request closed without merging**